src/helpers/serializer.ts

```diff
@@ -0,0 +1,179 @@
+import { DocumentSnapshot } from '@firebase/firestore-types'
+import { SerializerOptions } from 'src/types/Serializer'
+import { Fuego } from '../classes/Fuego'
+import {
+  CollectionQueryType,
+  SerializedCollectionQueryType,
+  WhereArray,
+  WhereItem,
+  WhereType,
+} from '../types/Query'
+
+export class Serializer {
+  private static snapshotCache: Record<string, DocumentSnapshot> = {}
+
+  private static isSnapshot(value: unknown, fuego: Fuego) {
+    if (!value) return false
+    return (
+      fuego.firestore?.DocumentSnapshot &&
+      value instanceof fuego.firestore.DocumentSnapshot
+    )
+  }
+
+  // Helper for where clauses
+  public static multipleConditions<Doc extends object = {}>(
+    w: WhereType<Doc>
+  ): w is WhereArray<Doc> {
+    return !!(w as WhereArray) && Array.isArray(w[0])
+  }
+
+  // Serializer function for where condition
+  private static serializeWhere<Doc extends object = {}>(
+    where: WhereType<Doc>,
+    fuego: Fuego
+  ): WhereType<Doc> {
+    if (this.multipleConditions(where)) {
+      return where.map(w => this.serializeWhere(w, fuego)) as WhereArray<Doc>
+    }
+    // Date: Inject serializer options if not specified
+    if (where[2] instanceof Date && !where[3]) {
+      return [...where.slice(0, 3), { type: 'date' }] as WhereItem<Doc>
```

Comment:
Shouldn't this turn the date into a serializable value, such as milliseconds or a string?

Reply:
Or I suppose JSON.stringify at the last step can handle Dates, never mind